frontend/src/feature/CreateTeamForm/model/api.ts

```diff
@@ -0,0 +1,92 @@
+import { queryOptions } from "@tanstack/react-query";
+import { getPlayersByYear, type Country } from "../../../entity";
+import { $api } from "../../../shared";
+import type { LeagueResponse, LineupResponse } from "./types";
+
+type GetAbortSignalProps = {
+  signal: AbortSignal;
+};
+
+export const getCountries = async ({
+  signal,
+}: GetAbortSignalProps): Promise<Country[] | undefined> => {
+  try {
+    const response = await $api.get("/countries", {
+      signal,
+    });
+
+    return response.data;
+  } catch (e) {
+    console.error("Error when try to fetch countries");
+  }
+};
+
+export const getLineups = async ({
+  signal,
+}: GetAbortSignalProps): Promise<LineupResponse[] | undefined> => {
+  try {
+    const response = await $api.get("/lineups", {
+      signal,
+    });
+
+    return response.data;
+  } catch (e) {
+    console.error("Error when try to fetch lineups");
+  }
+};
+
+export const getLeagues = async ({
+  signal,
+}: GetAbortSignalProps): Promise<LeagueResponse[] | undefined> => {
+  try {
+    const response = await $api.get("/leagues", {
+      signal,
+    });
+
+    return response.data;
+  } catch (e) {
+    console.error("Error when try to fetch leagues");
+  }
+};
+
+type GetPlayersQueryOptionsProps = {
```

Comment:
в таких случая лучше выносить типы в отдельный файл(чтобы не были в перемешку типы и функции)
мне не очень нравится нейминг аргументов этих функций, props это больше про компоненты
что-нибудь вроде 
```suggestion
type playersQueryOptions = {
```